Sources/PackageDescription/Context.swift

```diff
@@ -0,0 +1,30 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+/// The context a Swift package is running in. This encapsulates states that are known at build-time.
+/// For example where in the file system the current package resides.
+@available(_PackageDescription, introduced: 5.6)
+public struct Context {
+    private let model : ContextModel
+
+    public var packageDirectory : String {
+        model.packageDirectory
+    }
+    
+    public var environment : [String : String] {
+        model.environment
```

Comment:
Would it make sense for this to return a type that can just look up in the current environment rather than serializing the contents of the whole environment when invoking the manifest?

Reply:
Oh never mind, I didn't see that this is what ContextModel does.  That makes sense.